### PR TITLE
critest: Add test for exec

### PR DESCRIPTION
**Pull request by umohnani8**
Add test for exec where tty and stdin set to true

Signed-off-by: umohnani8 <umohnani@redhat.com>

### CODE REVIEW

**Comment by umohnani8** [comment]
@mrunalp PTAL

**Comment by umohnani8** [comment]
The exec test with stdin=true and tty=true is passing on cri-o, but fails on docker. Could be a bug in docker?

**Comment by feiskyer** [comment]
For stdin=true and tty=true, docker exec will stop as soon as stdin is closed (and so with containerd @Random-Liu). Since the input is an empty buffer, it is closed immediately, so the output streams are closed immediately, too.

@umohnani8 Could you keep the stdin open until getting the output in `checkExec()`?

**Comment by umohnani8** [comment]
@feiskyer how do I do that in the test? I thought the StreamOptions was accounting for that.

**Comment by feiskyer** [comment]
@umohnani8  Let me add a patch to your PR

File: pkg/validate/streaming.go [hunk 1]
```diff
@@ -119,39 +147,52 @@ var _ = framework.KubeDescribe("Streaming", func() {
...
+	By("exec given command in container: " + execReq.ContainerId)
+	resp, err := c.Exec(execReq)
+	framework.ExpectNoError(err, "failed to exec in container %q", execReq.ContainerId)
 	framework.Logf("Get exec url: " + resp.Url)
 	return resp.Url
 }
 
-func checkExec(c internalapi.RuntimeService, execServerURL string) {
+func checkExec(c internalapi.RuntimeService, execServerURL, stdout string, isTty bool) {
 	localOut := &bytes.Buffer{}
```
Comment (Random-Liu) [inline pkg/validate/streaming.go]: Use safe buffer. https://github.com/umohnani8/cri-tools/blob/094e42ff5cde024a4e6bae544c1993e0f88d83a6/pkg/validate/streaming.go#L236

`bytes.Buffer` is not thread safe.

Actually use similar code with `checkAttach`. https://github.com/umohnani8/cri-tools/blob/094e42ff5cde024a4e6bae544c1993e0f88d83a6/pkg/validate/streaming.go#L257

Comment (feiskyer) [reply pkg/validate/streaming.go]: Good catch 👍

**Comment by umohnani8** [comment]
@feiskyer I see now, thanks for the patch :)

**Comment by Random-Liu** [comment]
Actually we should test different behaviors when `stdin` and `tty` are true/false. Currently, the 2 test cases are exactly the same.

/lgtm for now.

**Pull request merged**